lib/widgets/compose_box.dart

```diff
@@ -1291,12 +1342,18 @@ class _ComposeBoxState extends State<ComposeBox> implements ComposeBoxState {
 
   @override
   Widget build(BuildContext context) {
-    final Widget body;
+    final Widget? body;
+    Widget? errorBanner;
 
-    final errorBanner = _errorBanner(context);
+    errorBanner = _errorBanner(context);
     if (errorBanner != null) {
-      body = errorBanner;
+      body = null;
```

Comment:
I feel that having to do this makes only calling `_ComposeBoxContainer` once less appealing.

Using an early return here when the `errorBanner` is present can make it easier to reason about the value of `body` and `errorBanner`. And we can comfortably pass `errorBanner: null` at the end, or `body: null` at the beginning, for clarity.

How about:
```diff
diff --git a/lib/widgets/compose_box.dart b/lib/widgets/compose_box.dart
index 1ca2e4ae..6514a331 100644
--- a/lib/widgets/compose_box.dart
+++ b/lib/widgets/compose_box.dart
@@ -1343,12 +1343,12 @@ class _ComposeBoxState extends State<ComposeBox> implements ComposeBoxState {
   @override
   Widget build(BuildContext context) {
     final Widget? body;
-    Widget? errorBanner;
 
-    errorBanner = _errorBanner(context);
+    final errorBanner = _errorBanner(context);
     if (errorBanner != null) {
-      body = null;
-    } else {
+      return _ComposeBoxContainer(body: null, errorBanner: errorBanner);
+    }
+
     // TODO(#720) dismissable message-send error, maybe something like:
     //     if (controller.sendMessageError.value != null) {
     //       errorBanner = _ErrorBanner(label:
@@ -1374,8 +1374,7 @@ class _ComposeBoxState extends State<ComposeBox> implements ComposeBoxState {
         assert(false);
         body = null;
     }
-    }
 
-    return _ComposeBoxContainer(body: body, errorBanner: errorBanner);
+    return _ComposeBoxContainer(body: body, errorBanner: null);
   }
 }
```